### PR TITLE
services.xserver.imwheel: Fix default extraOptions

### PR DESCRIPTION
**Pull request by turboMaCk**
###### Motivation for this change

related to https://github.com/NixOS/nixpkgs/pull/71052 it seems there was a mistake made during refactoring. `escapeShellArgs` wraps flags to `''` which results in imwheel not being able to parse them.

###### Things done
~Replace `escapeShellArgs` with `concatStringsSep " "` which fixes the problem.~
Change default `extraOptions` to be compatible with `escapeShellArgs`.

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @Infinisil sorry that this sliped.

### CODE REVIEW

**Comment by Infinisil** [comment]
`escapeShellArgs` should be preferrable because it escapes things properly. If you need to pass `--buttons 45` you can do so with either `[ "--buttons" "45" ]` or `[ "--buttons=45" ]`

**Comment by turboMaCk** [comment]
@Infinisil changed inlast commit though I have a feeling it might be quite unobvious behaviour for the user.

**Comment by Infinisil** [comment]
Hm yeah I know what you mean. Maybe it doesn't matter here since I don't think imwheel has any options that could need escaping (as of now anyways), but I think we should strive for proper escaping by default.

**Comment by turboMaCk** [comment]
I think in cases like this (and especially for service that won't be that widely used probably) the consistency is the most important thing. So if this is standard for services in nixpkgs then I belive it's right thing to do. I was more just making excuse why I instinctively removed the escaping rather then changed the defaults. Thanks for the feedback :100:

**Comment by turboMaCk** [comment]
> Maybe it doesn't matter here since I don't think imwheel has any options that could need escaping (as of now anyways)

I think there is still valid argument that this will teach folks to expect this format working across services and then in cases where service realy does need to escape they would shoot their foot which is not nice. I would rather guide towards single consistent behaviour apliable to all cases to avoid fragmentation which I belive is already a bit complicated to manage.

TL;DR: after thinking about it I think I was wrong and you're right.

**Review by infinisil — APPROVED** [review]
